src/server/transaction.cc

```diff
@@ -242,7 +245,13 @@ void Transaction::InitShardData(absl::Span<const PerShardCache> shard_index, siz
     unique_shard_id_ = i;
 
     for (size_t j = 0; j < si.args.size(); ++j) {
-      kv_args_.push_back(si.args[j]);
+      string_view arg = si.args[j];
+      kv_args_.push_back(arg);
+      if (si.key_step == 1 || j % si.key_step == 0) {
+        kv_fp_.push_back(LockTag(arg).Fingerprint());
+      } else {
+        kv_fp_.push_back(0);
+      }
```

Comment:
We can divide by key_step instead of storing 0s, if you don't like dividing, we can store the offsets separately.

The only condition is that we don't have bonus and key_step > 1 at the same time

Reply:
it will simplify all the loops in db_slice

Reply:
I prefer to keep key_step at least until we implement #2021 which requires a key step of 3.

Reply:
oh, I see. I do not know how to do it in a simple way. We need to know where each fingerprint slice starts  in GetLockArgs and it uses the regular key indices.

Reply:
ok, I know how to do it.